src/main/java/org/jenkinsci/backend/ircbot/IrcBotImpl.java

```diff
@@ -702,7 +702,9 @@ private GHTeam getOrCreateRepoLocalTeam(GHOrganization org, GHRepository r) thro
         if (t==null) {
             t = org.createTeam(teamName, Permission.ADMIN, r);
         } else {
-            t.add(r);
+            if (!t.getRepositories().containsValue(r)) {
```

Comment:
A comment here would be nice to have — it's not clear why adding a repo to a team again causes problems?  Or what a "legacy team" is?

Minor: I guess this could be `else if`.